code/test-utils/src/main/java/com/adobe/marketing/mobile/util/MockNetworkService.kt

```diff
@@ -0,0 +1,240 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.util
+
+import com.adobe.marketing.mobile.services.HttpConnecting
+import com.adobe.marketing.mobile.services.HttpMethod
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.NetworkCallback
+import com.adobe.marketing.mobile.services.NetworkRequest
+import com.adobe.marketing.mobile.services.Networking
+import com.adobe.marketing.mobile.services.TestableNetworkRequest
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+class MockNetworkService: Networking, TestResettable {
+    private val helper = TestNetworkService()
+    // Simulating the async network service
+    private val executorService: ExecutorService = Executors.newCachedThreadPool()
+
+    companion object {
+        private const val LOG_SOURCE = "MockNetworkService"
+        private var delayedResponse = 0
+        private val defaultResponse: HttpConnecting = object : HttpConnecting {
+            override fun getInputStream(): InputStream {
+                return ByteArrayInputStream("".toByteArray())
+            }
+
+            override fun getErrorStream(): InputStream? {
+                return null
+            }
+
+            override fun getResponseCode(): Int {
+                return 200
+            }
+
+            override fun getResponseMessage(): String {
+                return ""
+            }
+
+            override fun getResponsePropertyValue(responsePropertyKey: String): String? {
+                return null
+            }
+
+            override fun close() {}
+        }
+    }
+
+    override fun connectAsync(networkRequest: NetworkRequest, resultCallback: NetworkCallback?) {
+        Log.trace(
+            TestConstants.LOG_TAG,
+            LOG_SOURCE,
+            "Received connectUrlAsync to URL '%s' and HttpMethod '%s'.",
+            networkRequest.url,
+            networkRequest.method.name
+        )
```

Comment:
Thanks for the update, let's also take out the updates L73, L74 out of the executor block since the structures you are using (MutableMap, MutableList) are not thread safe, and keep only the response handler there.